From 372da54aea8dd3ec33318906827edf71508a3aa6 Mon Sep 17 00:00:00 2001
From: Zheng Yejian <zhengyejian1@huawei.com>
Date: Mon, 29 Aug 2022 20:03:38 +0000
Subject: [PATCH 0005/1076] [Huawei] livepatch/ppc64: Fix preemption check when
 enabling

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187321

--------------------------------

Misspelling of 'CONFIG_PREEMPTION' may cause old function not being
checked, which results in a running function being livepatched.

Fixes: 20106abf1e74 ("livepatch: Check whole stack when CONFIG_PREEMPT is set")
Signed-off-by: Zheng Yejian <zhengyejian1@huawei.com>
---
 arch/powerpc/kernel/livepatch_64.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/powerpc/kernel/livepatch_64.c b/arch/powerpc/kernel/livepatch_64.c
index 416f9f03d747..a2ec7c8c1bad 100644
--- a/arch/powerpc/kernel/livepatch_64.c
+++ b/arch/powerpc/kernel/livepatch_64.c
@@ -174,7 +174,7 @@ static int klp_check_activeness_func(struct klp_patch *patch, int enable,
 				 * excution of instructions to be repalced is
 				 * complete.
 				 */
-				if (IS_ENABLED(CONFIG_PREEMTION) ||
+				if (IS_ENABLED(CONFIG_PREEMPTION) ||
 				    (func->force == KLP_NORMAL_FORCE) ||
 				    check_jump_insn(func_addr)) {
 					ret = add_func_to_list(check_funcs, &pcheck,
-- 
2.17.1

